macOS nightly build tests are failing

<!-- Thanks for taking the time to report this build failure. To proceed with the report please:

1. Title the issue "Installation issue: <name-of-the-package>".
2. Provide the information required below.

We encourage you to try, as much as possible, to reduce your problem to the minimal example that still reproduces the issue. That would help us a lot in fixing it quickly and effectively! -->

### Steps to reproduce the issue

<!-- Fill in the exact spec you are trying to build and the relevant part of the error message -->
```console
$ spack install qt
```

with an external qt install as in our CI (#18642)
```
spack config add packages:opengl:paths:opengl@4.1:/usr/X11R6
```

fails with
```
2020-08-28T02:56:56.9860450Z [+] /Users/runner/work/spack/spack/opt/spack/darwin-catalina-x86_64/apple-clang-11.0.3/py-ipykernel-5.1.1-hxmeesoggzdzyhjczzdtfeguqpjzwk5k
2020-08-28T02:56:57.7652100Z ==> Installing qt
2020-08-28T02:56:57.7652340Z ==> No binary for qt found: installing from source
2020-08-28T02:58:40.9692410Z ==> Warning: microarchitecture specific optimizations are not supported yet on mixed compiler toolchains [check apple-clang@11.0.3 for further details]
2020-08-28T02:58:40.9694820Z ==> Warning: Copying Xcode from /Applications/Xcode_11.6.app to /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/runner/spack-stage/xcode-select/apple-clang/11.0.3 in order to add spack wrappers to it. Please do not interrupt.
2020-08-28T02:58:40.9789700Z ==> Error: AttributeError: Query of package 'opengl' for 'libs' failed
2020-08-28T02:58:40.9789900Z 	prefix : /usr/X11R6
2020-08-28T02:58:40.9791750Z 	spec : opengl@4.1%apple-clang@11.0.3 arch=darwin-catalina-x86_64
2020-08-28T02:58:40.9791950Z 	queried as : opengl
2020-08-28T02:58:40.9792060Z 	extra parameters : []
2020-08-28T02:58:40.9792120Z 
2020-08-28T02:58:40.9792240Z /Users/runner/work/spack/spack/lib/spack/spack/build_environment.py:863, in child_process:
2020-08-28T02:58:40.9792370Z         860            tb_string = traceback.format_exc()
2020-08-28T02:58:40.9792480Z         861
2020-08-28T02:58:40.9792620Z         862            # build up some context from the offending package so we can
2020-08-28T02:58:40.9792740Z   >>    863            # show that, too.
2020-08-28T02:58:40.9792860Z         864            package_context = get_package_context(tb)
2020-08-28T02:58:40.9792970Z         865
2020-08-28T02:58:40.9793080Z         866            build_log = None
2020-08-28T02:58:40.9793150Z 
2020-08-28T02:58:40.9793200Z 
2020-08-28T02:58:41.0313570Z ==> Warning: Skipping build of py-pyqt5 since qt failed
2020-08-28T02:58:41.0314200Z ==> Warning: Skipping build of py-qtconsole since py-pyqt5 failed
2020-08-28T02:58:41.0314670Z ==> Warning: Skipping build of py-jupyter since py-qtconsole failed
2020-08-28T02:58:41.0315160Z ==> Error: Terminating after first install failure: AttributeError: Query of package 'opengl' for 'libs' failed
2020-08-28T02:58:41.0315310Z 	prefix : /usr/X11R6
2020-08-28T02:58:41.0315750Z 	spec : opengl@4.1%apple-clang@11.0.3 arch=darwin-catalina-x86_64
2020-08-28T02:58:41.0315870Z 	queried as : opengl
2020-08-28T02:58:41.0316600Z 	extra parameters : []
2020-08-28T02:58:41.0316730Z /Users/runner/work/spack/spack/lib/spack/spack/build_environment.py:863, in child_process:
2020-08-28T02:58:41.0316860Z         860            tb_string = traceback.format_exc()
2020-08-28T02:58:41.0316970Z         861
2020-08-28T02:58:41.0317100Z         862            # build up some context from the offending package so we can
2020-08-28T02:58:41.0317220Z   >>    863            # show that, too.
2020-08-28T02:58:41.0317340Z         864            package_context = get_package_context(tb)
2020-08-28T02:58:41.0317450Z         865
2020-08-28T02:58:41.0317560Z         866            build_log = None
2020-08-28T02:58:41.0317620Z 
2020-08-28T02:58:41.0317720Z Traceback (most recent call last):
2020-08-28T02:58:41.0319350Z   File "/Users/runner/work/spack/spack/lib/spack/spack/build_environment.py", line 846, in child_process
2020-08-28T02:58:41.0319560Z     setup_package(pkg, dirty=dirty)
2020-08-28T02:58:41.0319780Z   File "/Users/runner/work/spack/spack/lib/spack/spack/build_environment.py", line 722, in setup_package
2020-08-28T02:58:41.0319930Z     set_build_environment_variables(pkg, build_env, dirty)
2020-08-28T02:58:41.0320140Z   File "/Users/runner/work/spack/spack/lib/spack/spack/build_environment.py", line 336, in set_build_environment_variables
2020-08-28T02:58:41.0320270Z     dep_link_dirs.extend(query.libs.directories)
2020-08-28T02:58:41.0320440Z   File "/Users/runner/work/spack/spack/lib/spack/spack/spec.py", line 900, in __get__
2020-08-28T02:58:41.0320590Z     raise AttributeError(message)
2020-08-28T02:58:41.0321120Z AttributeError: Query of package 'opengl' for 'libs' failed
2020-08-28T02:58:41.0321250Z 	prefix : /usr/X11R6
2020-08-28T02:58:41.0321700Z 	spec : opengl@4.1%apple-clang@11.0.3 arch=darwin-catalina-x86_64
2020-08-28T02:58:41.0321830Z 	queried as : opengl
2020-08-28T02:58:41.0321940Z 	extra parameters : []
2020-08-28T02:58:41.0321990Z 
2020-08-28T02:58:43.3941210Z ##[error]Process completed with exit code 1.
2020-08-28T02:58:43.4034300Z Post job cleanup.
2020-08-28T02:58:44.0585030Z [command]/usr/local/bin/git version
2020-08-28T02:58:44.0900940Z git version 2.28.0
2020-08-28T02:58:44.0975530Z [command]/usr/local/bin/git config --local --name-only --get-regexp core\.sshCommand
2020-08-28T02:58:44.1060350Z [command]/usr/local/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2020-08-28T02:58:44.2691900Z [command]/usr/local/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2020-08-28T02:58:44.2752070Z http.https://github.com/.extraheader
2020-08-28T02:58:44.2765750Z [command]/usr/local/bin/git config --local --unset-all http.https://github.com/.extraheader
2020-08-28T02:58:44.2855920Z [command]/usr/local/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2020-08-28T02:58:44.4345960Z Cleaning up orphan processes
```

### Information on your system

any recent macOS

### Additional information

See:
https://github.com/spack/spack/runs/1039421566?check_suite_focus=true

I think it's caused by this line:
  https://github.com/spack/spack/blob/25021ec2280cc61ec7e6ea7ea39dec773eb796c0/var/spack/repos/builtin/packages/qt/package.py#L272

because `.libs` does not exist for a virtual package ? (but for mesa)

### General information

<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->
- [ ] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [ ] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [ ] I have uploaded the build log and environment files
- [ ] I have searched the issues of this repo and believe this is not a duplicate

--- a/.github/workflows/macos_python.yml
+++ b/.github/workflows/macos_python.yml
@@ -46,8 +46,7 @@ jobs:
     - name: spack install
       run: |
         . .github/workflows/install_spack.sh
-        spack config add packages:opengl:paths:opengl@4.1:/usr/X11R6
-        spack install -v --fail-fast py-jupyter %apple-clang
+        spack install -v --fail-fast py-jupyterlab %apple-clang
 
   install_scipy_clang:
     name: scipy, mpl, pd

